### PR TITLE
[CDAP-13004] Adds PipelineDetailStore and PipelineDetail ActionCreator

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-13004

This is the first step in rewriting the Pipeline Detail view top panel in React (https://issues.cask.co/browse/CDAP-12903). In this PR I have added:
- PipelineDetailStore, which is the Redux rewrite of `pipeline-detail-runs-store.js` and `pipeline-detail-nonruns-store.js` merged together
- PipelineDetails Action Creator, which is the Redux rewrite of `pipeline-detail-actions.js`

I have not changed references to the old Angular stores and action creator yet. That will come in the next PR.

### CODE REVIEW

File: cdap-ui/app/cdap/components/PipelineDetails/store/ActionCreator.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+};
+
+const setMaxConcurrentRuns = (maxConcurrentRuns) => {
+  PipelineDetailStore.dispatch({
+    type: ACTIONS.SET_MAX_CONCURRENT_RUNS,
+    payload: { maxConcurrentRuns }
+  });
+};
+
+const pollRuns = (params) => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PipelineDetails/store/ActionCreator.js]: I see these poll functions, but where are the polls being stopped?

**Comment by tonybach** [comment]
@elfenheart Hmm I added that because we have a corresponding function in the Angular store (https://github.com/caskdata/cdap/blob/feature/rewrite-pipeline-detail-stores/cdap-ui/app/hydrator/services/detail/actions/pipeline-detail-actions.js#L37). 

This is used at https://github.com/caskdata/cdap/blob/feature/rewrite-pipeline-detail-stores/cdap-ui/app/hydrator/controllers/detail-ctrl.js#L68, where the function is just being called without the polls being stopped. This seems wrong, so I have removed the `pollRuns` function from my new store, and will address the issue in my next PR (https://github.com/caskdata/cdap/pull/9868).

**Comment by elfenheart** [comment]
So we should not poll from the resource standpoint anymore. In angular, we tie a polling function to `$scope`, so when the `$scope` is destroyed, the poll will be stopped automatically. 

In react, you should use Rx.Observable.interval to create a poll inside the component, that way you can unsubscribe and that will stop the poll when the component will unmount. So things like getStatistic should not be polling from action creator

File: cdap-ui/app/cdap/components/PipelineDetails/store/index.js [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+const pipelineDetails = (state = DEFAULT_PIPELINE_DETAILS, action = defaultAction) => {
+  switch (action.type) {
+    case ACTIONS.INITIALIZE_PIPELINE_DETAILS: {
+      let app = action.payload.app;
+      let appConfig = {...app};
+
+      appConfig.pipelineName = app.name;
+      appConfig.pipelineType = app.artifact.name;
+      appConfig.pipelineDescription = app.description;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PipelineDetails/store/index.js]: shouldn't need to duplicate these

File: cdap-ui/app/cdap/components/PipelineDetails/store/index.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+  });
+  return customConfig;
+};
+
+const DEFAULT_PIPELINE_DETAILS = {
+  // Pipeline level info
+  pipelineName: '',
+  pipelineType: '',
+  pipelineDescription: '',
+  cloneConfig: {},
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PipelineDetails/store/index.js]: should rename this cloneConfig to just config

File: cdap-ui/app/cdap/components/PipelineDetails/store/index.js [hunk 4]
```diff
@@ -69,6 +69,8 @@ const DEFAULT_PIPELINE_DETAILS = {
   name: '',
   type: '',
   description: '',
+  artifactVersion: '',
+  artifactScope: '',
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PipelineDetails/store/index.js]: wouldn't this be better to just store as the artifact full object? combine type, artifactVersion, and artifactScope?

File: cdap-ui/app/cdap/components/PipelineDetails/store/index.js [hunk 5]
```diff
@@ -101,10 +103,13 @@ const pipelineDetails = (state = DEFAULT_PIPELINE_DETAILS, action = defaultActio
       return {
         ...state,
         name: pipeline.name,
-        type: pipeline.artifact.name,
-        artifactVersion: pipeline.artifact.version,
-        artifactScope: pipeline.artifact.scope,
         description: pipeline.description,
+        artifact: {
+          ...state.artifact,
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/PipelineDetails/store/index.js]: this line is not needed

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Pull request merged**